src/test/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSpec.scala

```diff
@@ -201,25 +201,18 @@ class SubmissionSpec(_system: ActorSystem) extends TestKit(_system) with FlatSpe
     checkSubmissionStatus(workspaceService, data.submissionId)
   }
 
-  it should "return a successful Submission when given an entity expression that evaluates to an empty set of entites" in withWorkspaceService { workspaceService =>
+  it should "404 when given an entity expression that evaluates to an empty set of entities" in withWorkspaceService { workspaceService =>
```

Comment:
Should it? You're not looking anything up that might be Not Found. You might be able to argue a case for Bad Request or No Content but I'd be in favour of treating this normally and just returning a nothing.